### PR TITLE
JAMES-3438 James Webadmin CLI domain create command

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public Integer call() {
+        try {
+            DomainClient domainClient = Feign.builder()
+                .target(DomainClient.class, domainCommand.webAdminCli.jamesUrl + "/domains");
+            Response rs = domainClient.createADomain(domainName);
+            if (rs.status() == 204) {
+                return 0;
+            } else {
+                return 1;
```
Comment (mbaechler) [inline server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java:49]: magic number?

File: server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        } catch (Exception e) {
+            e.printStackTrace(domainCommand.err);
```
Comment (mbaechler) [inline server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java:52]: I don't think duping a stacktrace to the user is a good error reporting

Comment (chibenwa) [reply server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java:52]: What would you do instead? Logging?

File: server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java (hunk 1 above)
Comment (chibenwa) [reply server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java:49]: Here's the interpreted version :-) 

You can introduce a constant in order to give an understandable name to these numbers. (0, 1, 204)

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 3]
```diff
@@ -58,4 +58,38 @@ void domainListCommandShouldWShowOnlyDefaultDomain(GuiceJamesServer server) {
         assertThat(outputStreamCaptor.toString().trim().toCharArray()).containsOnly("localhost".toCharArray());
     }
 
+    @Test
+    void domainCreateCommandWithValidNameShouldSuccessfully(GuiceJamesServer server) {
+        Port port = server.getProbe(WebAdminGuiceProbe.class).getWebAdminPort();
+
+        int exitCode = WebAdminCli.executeFluent(new PrintStream(outputStreamCaptor), new PrintStream(errorStreamCaptor),
+            "--url", "http://127.0.0.1:" + port.getValue(), "domain", "create", "linagora.com");
+        assertThat(exitCode).isEqualTo(0);
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:67]: Group all of the assertions at the end of your test please, it makes it more readable.

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 4]
```diff
@@ -54,20 +54,22 @@ void domainListCommandShouldWShowOnlyDefaultDomain(GuiceJamesServer server) {
...
         Port port = server.getProbe(WebAdminGuiceProbe.class).getWebAdminPort();
 
         int exitCode = WebAdminCli.executeFluent(new PrintStream(outputStreamCaptor), new PrintStream(errorStreamCaptor),
             "--url", "http://127.0.0.1:" + port.getValue(), "domain", "create", "linagora.com");
-        assertThat(exitCode).isEqualTo(0);
 
         WebAdminCli.executeFluent(new PrintStream(outputStreamCaptor), new PrintStream(errorStreamCaptor),
             "--url", "http://127.0.0.1:" + port.getValue(), "domain", "list");
+
+        assertThat(exitCode).isEqualTo(WebAdminCli.CLI_FINISHED_SUCCEED);
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:72]: We tend to not reuse production constant in test (as modifying the constant itself will lead to an invalid result without breaking the test)

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java (hunk 3 above)
Comment (chibenwa) [reply server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java]: Not done ?

Comment (quantranhong1999) [reply server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java]: it's done

**Comment by Arsnael** [comment]
```
[a46f3f4a84f4ccec0d3c15d395a31a59e26a16e3] [ERROR] Failures: 
[a46f3f4a84f4ccec0d3c15d395a31a59e26a16e3] [ERROR]   CassandraSetMessagesMethodTest>SetMessagesMethodTest.sendingAMailShouldLeadToAppropriateMailboxCountersOnSent:1248 1 expectation failed.
[a46f3f4a84f4ccec0d3c15d395a31a59e26a16e3] JSON path [0][1].list[0].totalMessages doesn't match.
[a46f3f4a84f4ccec0d3c15d395a31a59e26a16e3] Expected: <1>
[a46f3f4a84f4ccec0d3c15d395a31a59e26a16e3]   Actual: 2
```
test this please

**Comment by chibenwa** [comment]
test this please

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]
the both no_review are based on #3917 right? Pay attention when merging, does not seem up-to-date with the upper PR

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**